x-pack/plugins/security_solution/server/endpoint/lib/files/check_deleted_files_task.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { throwUnrecoverableError } from '@kbn/task-manager-plugin/server';
+
+import type { EndpointAppContext } from '../../types';
+import {
+  fileIdsWithoutChunks,
+  getFilesByStatus,
+  updateFilesStatus,
+} from '../../services/actions/action_files';
+
+export const TYPE = 'endpoint:check-deleted-files-task';
+export const VERSION = '1.0.0';
+const TITLE = 'Security Solution Endpoint Deleted Files Periodic Tasks';
+const TIMEOUT = '2m';
+const SCOPE = ['securitySolution'];
+const INTERVAL = '5m';
+
+export interface CheckDeletedFilesTaskSetupContract {
+  endpointAppContext: EndpointAppContext;
+  core: CoreSetup;
+  taskManager: TaskManagerSetupContract;
+}
+
+export interface CheckDeletedFilesTaskStartContract {
+  taskManager: TaskManagerStartContract;
+}
+
+export class CheckDeletedFilesTask {
+  private logger: Logger;
+  private wasStarted: boolean = false;
+
+  constructor(setupContract: CheckDeletedFilesTaskSetupContract) {
+    const { endpointAppContext, core, taskManager } = setupContract;
+    this.logger = endpointAppContext.logFactory.get(this.getTaskId());
+
+    taskManager.registerTaskDefinitions({
+      [TYPE]: {
+        title: TITLE,
+        timeout: TIMEOUT,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask(taskInstance, core);
+            },
+            cancel: async () => {},
+          };
+        },
+      },
+    });
+  }
+
+  public start = async ({ taskManager }: CheckDeletedFilesTaskStartContract) => {
+    if (!taskManager) {
+      this.logger.error('missing required service during start');
+      return;
+    }
+
+    this.wasStarted = true;
+
+    try {
+      await taskManager.ensureScheduled({
+        id: this.getTaskId(),
+        taskType: TYPE,
+        scope: SCOPE,
+        schedule: {
+          interval: INTERVAL,
+        },
+        state: {},
+        params: { version: VERSION },
+      });
+    } catch (e) {
+      this.logger.debug(`Error scheduling task, received ${e.message}`);
```

Comment:
in this case here - should we instead log to `.error()`? because `.debug()` is not seen until someone turns that on.

Also wondering if you should just log the entire error object `e`?